docs/smart/principals.md

```diff
@@ -13,47 +13,60 @@ _Principals_ are a Clarity native type that represents a spending entity. This s
+type that transfers `amount` uSTX from the sender to a recipient if some
+condition is met, otherwise returning a 400 error code.
 
-A given principal operates on its assets by issuing a signed transaction on the Stacks blockchain. A Clarity contract can use a globally defined `tx-sender` variable to obtain the current principal.
-
-The following user-defined function transfers an asset, in this case, tokens, between two principals:
-
-```
-(define (transfer! (sender principal) (recipient principal) (amount int))
-  (if (and
-        (not (eq? sender recipient))
-        (debit-balance! sender amount)
-        (credit-balance! recipient amount))
-    true
-    false))
+```cl
+(define-public (transfer-to-recipient! (recipient principal) (amount uint))
+  (if (check-condition)
+      (stx-transfer? amount tx-sender recipient)
+      (err u400)))
 ```
 
-The principal's signature is not checked by the smart contract, but by the virtual machine.
-
-
 ## Smart contracts as principals
 
-Smart contracts themselves are principals and are represented by the smart contract's identifier. You create the identifier when you launch the contract, for example, the contract identifier here is `hanomine`.
+Smart contracts themselves are principals and are represented by the
+smart contract's identifier -- which is the publishing address of the
+contract _and_ the contract's name, e.g.:
 
-```bash
-clarity-cli launch hanomine /data/hano.clar /data/db
+```cl
+'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR.contract-name
 ```
 
-A smart contract may use the special variable `contract-name` to refer to its own principal.
+For convenience, smart contracts may write a contract's identifier in the
+form `.contract-name`, which is expanded by the Clarity interpreter into
+a fully-qualified contract identifier that corresponds to the same
+publishing address as the contract it appears in.
 
-To allow smart contracts to operate on assets it owns, smart contracts may use the special `(as-contract expr)` function. This function executes the expression (passed as an argument) with the `tx-sender` set to the contract's principal, rather than the current sender. The `as-contract` function returns the value of the provided expression.
+In order for a smart contract to operate on assets it owns, smart contracts
+may use the special `(as-contract ...)` function. This function
+executes the expression (passed as an argument) with the `tx-sender`
+set to the contract's principal, rather than the current sender. The
+`as-contract` function returns the value of the provided expression.
 
-For example, a smart contract that implements something like a "token faucet" could be implemented as so:
+For example, a smart contract that implements something like a "token
+faucet" could be implemented as so:
 
 ```cl
 (define-public (claim-from-faucet)
-  (if (is-none? (fetch-entry claimed-before (tuple (sender tx-sender))))
+  (if (is-none? (fetch-entry claimed-before {sender: tx-sender}))
```

Comment:
If we're going to use `{ }` in examples, it should be introduced earlier as the way to define tuples.

Reply:
There's no documentation for how to declare any literals in these documents at the moment. That could be added as a section to the clarityRef.md file, I think.

Reply:
Replaced with the explicit tuple syntax in https://github.com/blockstack/stacks-blockchain/pull/1439/commits/9eb9677a3433ce0d0949dd085ccff96b39976aec